nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/test/java/org/nuxeo/ecm/platform/comment/CommentUtils.java

```diff
@@ -0,0 +1,140 @@
+     * 
+     * @param mails the current mails
+     * @param commentDocModel the document model of the comment, cannot be {@code null}
+     * @param commentedDocModel the document being commented, cannot be {@code null}
+     * @param event the event, cannot be {@code null}
+     * @param commentEventType the type of comment event {@link org.nuxeo.ecm.platform.comment.api.CommentEvents},
+     *            cannot be {@code null}
+     * @param expectedMailsCount expected received mails count
+     */
+    public static void checkReceivedMail(List<MailMessage> mails, DocumentModel commentDocModel,
+            DocumentModel commentedDocModel, Event event, String commentEventType, int expectedMailsCount) {
+        assertEquals(expectedMailsCount, mails.size());
+
+        if (expectedMailsCount > 0) {
+            String subject = String.format("[Nuxeo5]%s comment on '%s'",
+                    COMMENT_ADDED.equals(event.getName()) ? "New" : "Updated", commentedDocModel.getName());
+            List<MailMessage> mailsBySubject = mails.stream()
+                                                    .filter(m -> subject.equals(m.getSubject()))
+                                                    .collect(Collectors.toList());
+            assertEquals(1, mailsBySubject.size());
+
+            String expectedMailContent = getExpectedMailContent(commentDocModel, commentedDocModel, event,
+                    commentEventType);
+            assertEquals(expectedMailContent, getMailContent(mailsBySubject.get(0)));
+        }
+    }
+
+    public static String getExpectedMailContent(DocumentModel commentDocModel, DocumentModel commentedDocModel,
+            Event event, String commentEventType) {
+        String date = EVENT_DATE_FORMAT.format(Date.from(Instant.ofEpochMilli(event.getTime())));
+        StringBuilder builder = new StringBuilder(
+                "(").append(commentDocModel.getPropertyValue(COMMENT_AUTHOR))
+                    .append(")")
+                    .append(" has")
+                    .append(COMMENT_ADDED.equals(commentEventType) ? " added" : " updated")
+                    .append(" a comment on ")
+                    .append(commentedDocModel.getName())
+                    .append(" at ")
+                    .append(date)
+                    .append(". says: ")
+                    .append(commentDocModel.getPropertyValue(COMMENT_TEXT))
+                    .append(" » See all the comments of the document")
+                    .append(" You received this notification because you subscribed to 'New comment' notification on this document or on one of its parents.");
+
+        return builder.toString();
+    }
+
+    public static String getMailContent(MailMessage mailMessage) {
+        String content = mailMessage.getContent();
+        if (mailMessage.getContentType().contains("text/html")) {
+            Document document = Jsoup.parse(content);
+            return document.select("body").text();
```

Comment:
i saw some others alternatives and initially i was going to use Nuxeo `Html2TextConverter` and after a discussion with Kevin we took the choice to use the library that is used by this converter instead of the whole converter fyi the library is http://jericho.htmlparser.net/docs/index.html